substrings.rb

```diff
@@ -0,0 +1,8 @@
+def substrings(string, dictionary)
+    count_hash = {}
+    dictionary.map {|substring| count_hash[substring] = string.downcase.scan(/#{substring.downcase}/).count }
```

Comment:
1. do we need inside the loop each time to do `string.downcase`?
2. it's better to put less logic inside each block - so for example you can first go over all the values, turn them all downcase, then do the second action on them
3. `#map` should be used if you are returning value - otherwise use `#each`
4. when your doing array.count - it's counting (going one by one) - see https://apidock.com/ruby/Enumerable/count. you should use `.length` - it gets the array length property
```suggestion
    string.downcase!
    dictionary
      .map(&:downcase)
      .each {|substring| count_hash[substring] = string.scan(/#{substring}/).length }
```
5. in the exercise it was written
> It should return a hash listing each substring (case insensitive) that was found

but - you are returning also substrings that are not found with 0 as the value

Reply:
1. done
2. I don't think the `&:` syntax works in `each_with_object`
3. not relevant with change from previous comment
4. done
5. added the condition